proactive-jupyter-kernel/kernel.py

```diff
@@ -1971,9 +1992,96 @@ def __submit_job__(self, input_data):
 
     def __list_submitted_jobs__(self, input_data):
         for job_id in self.submitted_jobs_names:
-            self.__kernel_print_ok_message__('Id: ' + str(job_id) + ' , Name: ' + self.submitted_jobs_names[job_id]
-                                             + '\n')
+            self.__kernel_print_ok_message__('Id: ' + str(job_id) + ' , Name: ' + self.submitted_jobs_names[job_id]+ '\n')
 
+    def __list_html_table__(self, input_data):
+        html = ''
+        for d in input_data:
+            html += '<tr>' + ''.join('<td>' + d + '</td>') + '</tr>'
+        return html
+
+    def __display_html__(self, input_data):
+        content = {'data': {'text/html':  input_data}, 'metadata': {}}
+        self.send_response(self.iopub_socket, 'display_data', content)
+
+    def __get_target_env_nodesources__(self, input_data, send_response=True):
+        restapi = self.gateway.getProactiveRestApi()
+        nodesources = restapi.get_rm_model_nodesources()
+        
+        html = self.__list_html_table__(nodesources)   
+        if html =='':
+            warning_msg = '<div><i class="fa fa-warning"></i> Warning Message: None of node source is available.</div>'
+            if send_response:
+                self.__display_html__(warning_msg)
+            return  html
+        else:
+            html = '<table border=2 class="stocktable" id="table1" > <th>List of available node sources</th>' + html + '</table>'
+
+            if send_response:
+                html = html + '<div><i class="fa fa-info-circle"></i> If you don\'t set a node source name for your job, it will run on any available node source.</div>'
+                self.__display_html__(html)
+            else:
+                return html
+
+    def __get_target_env_hosts__(self, input_data, send_response=True):
+        restapi = self.gateway.getProactiveRestApi()
+        hosts = restapi.get_rm_model_hosts()
+
+        html = self.__list_html_table__(hosts)   
+        if html =='':
+            warning_msg = '<div><i class="fa fa-warning"></i> Warning Message: None of node source is available.</div>'
+            if  send_response:
+                self.__display_html__(warning_msg)
+            return  html
+        else:
+            html = '<table border=2 class="stocktable" id="table1" > <th>List of available hots</th>' + html + '</table>'
+
+            if send_response:
+                html = html + '<div class="info-msg"><i class="fa fa-info-circle"></i> If you don\'t set a host name for your job, it will run on any available host.</div>'
+                self.__display_html__(html)
+            else:
+                return html     
+
+    def __get_target_env_tokens__(self, input_data, send_response=True, warning_response=True):
```

Comment:
I don't see the point of listing all tokens without any information regarding the related host or NS.